test/e2e/scale.go

```diff
@@ -333,6 +337,49 @@ func scaleSpec(ctx context.Context, inputGetter func() scaleSpecInput) {
 			Fail("")
 		}
 
+		if !input.SkipUpgrade {
+			By("Upgrade the workload clusters concurrently")
+			// Use the correct creator function for creating the workload clusters.
+			// Default to using the "create and wait" creator function. If SkipWaitForCreation=true then
+			// use the "create only" creator function.
+			upgrader := getClusterUpgradeAndWaitFn(framework.UpgradeClusterTopologyAndWaitForUpgradeInput{
+				ClusterProxy:                input.BootstrapClusterProxy,
+				KubernetesUpgradeVersion:    input.E2EConfig.GetVariable(KubernetesVersionUpgradeTo),
+				EtcdImageTag:                input.E2EConfig.GetVariable(EtcdVersionUpgradeTo),
+				DNSImageTag:                 input.E2EConfig.GetVariable(CoreDNSVersionUpgradeTo),
+				WaitForMachinesToBeUpgraded: input.E2EConfig.GetIntervals(specName, "wait-machine-upgrade"),
```

Comment:
Do we also need WaitForMachinePoolToBeUpgraded?

Reply:
(see below)

Reply:
Answered below - will close this comment to keep the conversation in one place.